### DISCUSSION TITLE
How do I use a command line flag to configure the log level for tauri-plugin-log?

### DISCUSSION THREAD

**Discussion by Raphiiko** [tauri-apps/tauri#11592] [Q&A] [answered]
I'm looking to let users configure the log level of my Tauri (v2) application by passing a command line flag. Ideally the log level for this would be configurable, but I'm just trying to use a `--debug` flag for now.

To do this, I'd like to use the [cli](https://v2.tauri.app/plugin/cli/) plugin. However, I'm running into a roadblock I'm not sure how to get around. Please see the following simplified code snippet:

```rust
pub fn run() {
    let mut builder = tauri::Builder::default()
        .plugin(tauri_plugin_cli::init()) // Configure the cli plugin
        .plugin(
            tauri_plugin_log::Builder::new() // Configure the log plugin
                .level(log::LevelFilter::Info) // Configure the log level (How do I change this based on a command line flag?)
                .build(),
        )
        .setup(move |app| {
            // Only now I can access the app handle to access the CLI matches
            let matches = app.cli().matches(); 
            // But by now the log plugin is already configured...
            Ok(())
        })
        .invoke_handler(tauri::generate_handler![...])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

According to the cli plugin's [docs](https://v2.tauri.app/plugin/cli/), I've only got access to the command line arguments once I've got a reference to the `App` struct (in the `setup` hook). However, the log level for the log plugin is already configured in the `plugin` call, where I have no access to the `App` struct just yet.

Does anyone know how I could make this work?

**Comment by FabianLars** [comment, marked as answer]
i'd probably use [clap](https://docs.rs/clap/latest/clap/) (or a similar crate) directly instead of the cli plugin. The cli plugin is imo only worth if you need the js apis.

**Reply by Raphiiko** [reply]
Check. I assumed I was missing something obvious, but if that's its main use case it makes more sense. I'll continue from there, thanks!
